torch/nn/modules/module.py

```diff
@@ -24,6 +24,12 @@
 T = TypeVar('T', bound='Module')
 
 
+def _get_special_module_supported_device_types():
```

Comment:
I don't think we want this function. "special" is not really a helpful concept here and each Module code needs custom handling of devices.
In this context, adding the explicit list everywhere sounds best.

Reply:
> I don't think we want this function. "special" is not really a helpful concept here and each Module code needs custom handling of devices.
> In this context, adding the explicit list everywhere sounds best.

yeah I have made changes to add custom device type directly and you could review again @albanD